cycode/cli/files_collector/iac/tf_content_generator.py

```diff
@@ -44,14 +43,23 @@ def _generate_tf_content(resource_changes: List[ResourceChange]) -> str:
 
 
 def _generate_resource_content(resource_change: ResourceChange) -> str:
-    resource_content = f'resource "{resource_change.resource_type}" "{resource_change.name}-{uuid.uuid4()}" {{\n'
+    resource_content = f'resource "{resource_change.resource_type}" "{_get_resource_name(resource_change)}" {{\n'
     if resource_change.values is not None:
         for key, value in resource_change.values.items():
             resource_content += f'  {key} = {json.dumps(value)}\n'
     resource_content += '}\n\n'
     return resource_content
 
 
+def _get_resource_name(resource_change: ResourceChange) -> str:
+    parts = [resource_change.module_address, resource_change.name]
+
+    if resource_change.index is not None:
+        parts.append(str(resource_change.index))
+
+    return '-'.join(filter(None, parts)).replace('.', '-')
```

Comment:
could you pls explain trick around `filter(None, someList`)?

Reply:
I just wanted to concat only the parts that are not None

Reply:
understood. IMO `[part for part in parts if part]` could be more readable because i had no clue that filter accepts `None` to the `fun` param. but as you want. maybe just add some comment around is enough

Reply:
changed per your suggestion